Fix HTTP credentials, allowing JSON objects as form inputs

taylordowns2000 opened the pull request:
## Description

This PR fixes #3437 , allowing the new HTTP configuration schema (and anything that asks for a raw JSON object in the future) to be used in Lightning credentials.

## Validation steps

1. Try to create an HTTP credential. (It shouldn't crash.)
2. Try to pass a valid JSON object in for tls options.
3. Try to pass an invalid JSON object for tls options.
4. Save, then run a job that logs the `state.configuration` to check it's all saved properly. (Or fetch the credential body via IEX.)

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [x] Code generation (copilot but not intellisense)
- [x] Learning or fact checking
- [x] Strategy / design
- [x] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

midigofrank reviewed (approved):
Nicely done 🙌

elias-ba commented on lib/lightning/credentials/schema.ex:
```diff
@@ -126,6 +126,18 @@ defmodule Lightning.Credentials.Schema do
         Changeset.add_error(changeset, field, "can't be blank")
+
+      %{expected: ["object"], actual: "string"} ->
+        # Check if the value is actually a valid JSON object
```
Can we remove this comment ?

taylordowns2000 replied:
Why?

elias-ba reviewed (approved):
Hey @taylordowns2000 this is looking good. Thanks 👏🏽 

PS: I have just one tiny tiny change request about an unnecessary comment but that's not gonna block this PR.

elias-ba commented:
Hey @taylordowns2000, I did approve this PR but then I saw CodeCov complaining about the coverage so I pushed a commit to help out with the test coverage and fixed a bug I found along the way.

The code would crash when someone passed non-string values (like integers or atoms) for object fields. ExJsonSchema returns different error types like `%{expected: ["object"], actual: "integer"}` that weren't handled. Added a catch-all case for those.

I also added few tests for the `validate_json_object/1` function.

I also refactored that big case statement, Credo was complaining about complexity, so I split it into multiple function clauses.

Everything's looking great. Let me know what you think!

taylordowns2000 commented:
> Hey @taylordowns2000, I did approve this PR but then I saw CodeCov complaining about the coverage so I pushed a commit to help out with the test coverage and fixed a bug I found along the way.
> 
> The code would crash when someone passed non-string values (like integers or atoms) for object fields. ExJsonSchema returns different error types like `%{expected: ["object"], actual: "integer"}` that weren't handled. Added a catch-all case for those.
> 
> I also added few tests for the `validate_json_object/1` function.
> 
> I also refactored that big case statement, Credo was complaining about complexity, so I split it into multiple function clauses.
> 
> Everything's looking great. Let me know what you think!

looks great @elias-ba ! assuming you've done extensive UI click testing here, i'm happy to merge. can you confirm that you've tested not only the changed credential type, but also some basic regression testing on other credential field types? i want to make sure that this big refactor doesn't mess anything up! once you confirm that, let's merge!
